### PR TITLE
Add Azure resource detector

### PR DESCRIPTION
**Pull request by RassK**
## Why

Towards #2395 

## What

* Adds resource detector for Azure App Service

## Tests

Extended.

## Checklist

- [x] `CHANGELOG.md` is updated.
- [x] Documentation is updated.
- [x] New features are covered by tests.

### CODE REVIEW

**Review by nrcventura — APPROVED** [review]

**Comment by RassK** [comment]
SIG: Waiting for next version (big changes are coming)

File: docs/config.md [hunk 1]
```diff
@@ -86,9 +86,10 @@ for more details.
...
+|-------------|----------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------|
+| `CONTAINER` | Container detector         | [Container resource detector documentation](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/ResourceDetectors.Container-1.0.0-beta.3/src/OpenTelemetry.ResourceDetectors.Container/README.md) | [Experimental](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md) |
+| `Azure`     | Azure App Service detector | [Container resource detector documentation](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/ResourceDetectors.Azure-1.0.0-alpha.1/src/OpenTelemetry.ResourceDetectors.Azure/README.md)        | [Experimental](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md) |
```
Comment (Kielek) [inline docs/config.md:92]:
```suggestion
| `AZURE`     | Azure App Service detector | [Container resource detector documentation](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/ResourceDetectors.Azure-1.0.0-alpha.1/src/OpenTelemetry.ResourceDetectors.Azure/README.md)        | [Experimental](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md) |
```

**Comment by rajkumar-rangaraj** [comment]
`OpenTelemetry.ResourceDetectors.Azure` is  `1.0.0-beta.1` we could proceed with PR.

**Comment by Kielek** [comment]
Azure detecor package references `System.Text.Json` 7.0.2.
Packages from main repository targeting 4.7.2. Can we downgrade it in contrib package?

@rajkumar-rangaraj?

File: docs/config.md [hunk 2]
```diff
@@ -88,9 +88,10 @@ for more details.
...
+|-------------------|----------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------|
+| `CONTAINER`       | Container detector         | [Container resource detector documentation](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/ResourceDetectors.Container-1.0.0-beta.3/src/OpenTelemetry.ResourceDetectors.Container/README.md) | [Experimental](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md) |
+| `AZUREAPPSERVICE` | Azure App Service detector | [Container resource detector documentation](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/ResourceDetectors.Azure-1.0.0-beta.1/src/OpenTelemetry.ResourceDetectors.Azure/README.md)         | [Experimental](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md) |
```
Comment (Kielek) [inline docs/config.md:94]:
```suggestion
| `AZUREAPPSERVICE` | Azure App Service detector | [Container resource detector documentation](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/ResourceDetectors.Azure-1.0.0-beta.2/src/OpenTelemetry.ResourceDetectors.Azure/README.md)         | [Experimental](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md) |
```

File: src/OpenTelemetry.AutoInstrumentation/Configurations/ResourceConfigurator.cs [hunk 3]
```diff
@@ -40,7 +41,8 @@ public static ResourceBuilder CreateResourceBuilder(IReadOnlyList<ResourceDetect
             resourceBuilder = enabledResourceDetector switch
             {
                 ResourceDetector.Container => Wrappers.AddContainerResourceDetector(resourceBuilder),
-                _ => resourceBuilder,
+                ResourceDetector.AzureAppService => Wrappers.AddAzureAppServiceResourceDetector(resourceBuilder),
```
Comment (rajkumar-rangaraj) [inline src/OpenTelemetry.AutoInstrumentation/Configurations/ResourceConfigurator.cs:44]: We could also include `AzureVMResourceDetector` it's a part of same package.

Ref: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/main/src/OpenTelemetry.ResourceDetectors.Azure#vm-resource-detector

**Review by rajkumar-rangaraj — APPROVED** [review]

**Review by nrcventura — APPROVED** [review]

**Review by pjanotti — APPROVED** [review]

**Pull request merged**